modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -56,4 +60,85 @@ object SmartGcalDao {
     } yield r
   }
 
+  private def lookup(step: Option[Step[InstrumentConfig]], loc: Location.Middle): ConnectionIO[LookupResult] = {
```

Comment:
These computations might be easier to express in terms of `EitherT`, which would let you avoid some of the `.left/.right/.fold/.point`ing, but it's fine as-is. Just something to keep in mind.